src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -396,29 +396,14 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         path_str: &'a str,
         module_id: DefId,
     ) -> Result<Res, ResolutionFailure<'a>> {
-        let path = ast::Path::from_ident(Ident::from_str(path_str));
         self.cx.enter_resolver(|resolver| {
-            // FIXME(jynelson): does this really need 3 separate lookups?
-            if let Ok((Some(ext), res)) = resolver.resolve_macro_path(
-                &path,
-                None,
-                &ParentScope::module(resolver.graph_root(), resolver),
-                false,
-                false,
```

Comment:
perhaps we should also add tests that ensure that macros can still be linked to on both newer and older editions

Reply:
Ah, yeah, you noted a problem with `#[macro_export]` above

Reply:
Yeah this code was like this to handle `macro_export`/old-style macro resolution. Which still exists. But perhaps we can handle this a bit cleaner.